src/main/java/mate/jdbc/dao/impl/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    private static final String ID = "id";
+    private static final String NAME = "name";
+    private static final String COUNTRY = "country";
+    private static final String CREATE_STATEMENT =
+            "INSERT INTO manufacturers (name, country) VALUES (?, ?);";
+    private static final String GET_STATEMENT = "SELECT * FROM manufacturers WHERE id = ?;";
+    private static final String GET_ALL_STATEMENT = "SELECT * FROM manufacturers;";
+    private static final String UPDATE_STATEMENT =
+            "UPDATE manufacturers SET name = ?, country = ? WHERE id = ?;";
+    private static final String DELETE_STATEMENT =
+            "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createFormatStatement =
+                        connection.prepareStatement(CREATE_STATEMENT,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createFormatStatement.setString(1, manufacturer.getName());
+            createFormatStatement.setString(2, manufacturer.getCountry());
+            createFormatStatement.executeUpdate();
+            ResultSet generatedKeys = createFormatStatement.getGeneratedKeys();
+
+            if (generatedKeys.next()) {
+                manufacturer.setId(generatedKeys.getLong(1));
+            }
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create manufacturer", e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getStatement = connection.prepareStatement(GET_STATEMENT)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            return resultSet(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all formats from DB", e);
+        }
+    }
+
+    private Optional<Manufacturer> resultSet(ResultSet resultSet) {
```

Comment:
Please rework method so it can be used in both get() and getAll() methods
```suggestion
    private Manufacturer getManufacturerFromResultSet(ResultSet resultSet) {
```